newsroom/utils.py

```diff
@@ -515,3 +533,99 @@ def deep_get(val: Dict[str, Any], keys: str, default: Optional[Any] = None) -> A
         keys.split("."),
         val,
     )
+
+
+def split_words(text: str) -> List[str]:
+    """Get word count for given plain text.
+
+    Any changes to this code **must** be reflected in the typescript version as well
+    Copied from ``superdesk-core:superdesk/text_utils.py:get_text_word_count``
+    """
+
+    flags = regex.MULTILINE | regex.UNICODE
+    initial_text_trimmed = text.strip()
+
+    if len(initial_text_trimmed) < 1:
+        return []
+
+    r0 = get_text(initial_text_trimmed, space_on_elements=True)
+    r1 = regex.sub(r"\n", " ", r0, flags=flags)
+
+    # Remove spaces between two numbers
+    # 1 000 000 000 -> 1000000000
+    r2 = regex.sub(r"([0-9]) ([0-9])", "\\1\\2", r1, flags=flags)
+
+    # remove anything that is not a unicode letter, a space, comma or a number
+    r3 = regex.sub(r"[^\p{L} 0-9,]", "", r2, flags=flags)
+
+    # replace two or more spaces with one space
+    r4 = regex.sub(r" {2,}", " ", r3, flags=flags)
+
+    return r4.strip().split(" ")
+
+
+def short_highlighted_text(html: str, max_length: int = 40, output_html: bool = True) -> str:
```

Comment:
I agree using ES to do this for us would be faster, but would also be inconsistent with what we offer in the front-end. 

It may create confusion as to why shortening/highlighting of text behaves differently in the browser vs an email.

Reply:
yep we might revisit this later